adapters/repos/db/node_wide_metrics.go

```diff
@@ -302,3 +317,131 @@ func (o *nodeWideMetricsObserver) Usage(filter tenantactivity.UsageFilter) tenan
 		return o.lastTenantUsage
 	}
 }
+
+// ----------------------------------------------------------------------------
+// Vector dimensions tracking
+// ----------------------------------------------------------------------------
+
+// Start a goroutine to collect vector dimension/segment metrics from the shards,
+// and publish them at a regular interval. Only call this method in the constructor,
+// as it does not guard access with locks.
+// If vector dimension tracking is disabled, this method is a no-op: no goroutine will
+// be started and the "done" channel stays nil.
+func (o *nodeWideMetricsObserver) observeDimensionMetrics() {
+	enterrors.GoWrapper(func() {
+		interval := config.DefaultTrackVectorDimensionsInterval
+		if o.db.config.TrackVectorDimensionsInterval > 0 { // duration must be > 0, or time.Timer will panic
+			interval = o.db.config.TrackVectorDimensionsInterval
+		}
+
+		// This is a low-priority background process, which is not time-sensitive.
+		// Some downstream calls require a context, so we create one, but we needn't
+		// manage it beyond making sure it doesn't leak.
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		o.publishVectorMetrics(ctx)
+
+		tick := time.NewTicker(interval)
+		defer tick.Stop()
+
+		for {
+			select {
+			case <-o.shutdown:
+				return
+			case <-tick.C:
+				o.publishVectorMetrics(ctx)
+			}
+		}
+	}, o.db.logger)
+}
+
+func (o *nodeWideMetricsObserver) publishVectorMetrics(ctx context.Context) {
+	// We're a low-priority process, copy the index map to avoid blocking others.
+	// No new indices can be added while we're holding the lock anyways.
+	o.db.indexLock.RLock()
+	indices := make(map[string]*Index, len(o.db.indices))
+	maps.Copy(indices, o.db.indices)
+	o.db.indexLock.RUnlock()
+
+	var total DimensionMetrics
```

Comment:
we will always start counting from 0 yes? no need for explicit setting total to 0?